pkg/apis/agones/v1/gameserver.go

```diff
@@ -438,6 +451,23 @@ func (gss *GameServerSpec) Validate(apiHooks APIHooks, devAddress string) ([]met
 		}
 	}
 
+	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
```

Comment:
To avoid the cyclic complexity, I'd suggest pulling this into it's own helper function.